commands/cli/helptext.go

```diff
@@ -351,18 +351,25 @@ func subcommandText(cmd *cmds.Command, rootName string, path []string) []string
 	if len(path) > 0 {
 		prefix += " "
 	}
+
+	// Sorting fixes changing order bug #2981.
+	sortedNames := make([]string, 0)
+	for name := range cmd.Subcommands {
+		sortedNames = append(sortedNames, name)
+	}
+	sort.Strings(sortedNames)
+
 	subcmds := make([]*cmds.Command, len(cmd.Subcommands))
 	lines := make([]string, len(cmd.Subcommands))
 
-	i := 0
-	for name, sub := range cmd.Subcommands {
+	for i, name := range sortedNames {
+		sub := cmd.Subcommands[name]
```

Comment:
Is it?
I ran `gofmt -l` and it didn't report any errors.
![out8167](https://cloud.githubusercontent.com/assets/1278705/17363558/7918a228-597b-11e6-8664-2729f2d61601.png)